pkg/handler/repositories.go

```diff
@@ -131,12 +140,19 @@ func (rh *RepositoryHandler) bulkCreateRepositories(c echo.Context) error {
 		newRepositories[i].FillDefaults()
 	}
 
-	response, err := rh.RepositoryDao.BulkCreate(newRepositories)
-	if err != nil {
-		return c.JSON(httpCodeForError(err), response)
+	responses, errs := rh.RepositoryDao.BulkCreate(newRepositories)
+	if len(errs) > 0 {
+		errorResponses := make([]api.ErrorResponse, len(errs))
+		for i := 0; i < len(errs); i++ {
+			if errs[i] != nil {
+				errorResponses[i] = api.NewErrorResponse(httpCodeForError(errs[i]), "Error creating repository", errs[i].Error())
+			} else {
+				errorResponses[i] = api.ErrorResponse{}
+			}
+		}
+		return c.JSON(http.StatusOK, echo.Map{"errors": errorResponses})
```

Comment:
I made a comment yesterday, but I've been trying to change the structure to this to better accommodate this scenario.
```
type ErrorResponse struct {
    Errors []Error{ Status, Detail, Title }
}
```

If I can't get this to work (right now it's tricky to avoid circular dependencies), I think I will try your first suggestion.

Reply:
The idea with this being I can just return ErrorResponse from the handler, which contains 1 or more errors. Then the error handler figures out what to do with it.

The error response creation function would look like your second suggest, where a list of errors is passed.

Reply:
sorry I didn't see the comment :+1:

About the circular references; I have turned some ideas in my mind, but still I don't fit all the pieces; I have some resources in mind, and I have to share them.

Reply:
I think I'll have to make a new package to put some of this new code. I don't see a way to fit it all together with the existing package structure.

Reply:
I'll do that and we can see from there